### PR TITLE
Additional user info

### PR DESCRIPTION
**Pull request by szok**
Added:
- Table renderer mechanism
- Additional user info

Connected to #1696

### CODE REVIEW

File: src/ralph/lib/table/templatetags/table.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from django import template
+from ralph.admin.helpers import get_field_by_relation_path
+
+register = template.Library()
+
+
+@register.filter
+def get_value(instance, field):
```
Comment (mkurek) [inline src/ralph/lib/table/templatetags/table.py]: missing docstring

File: src/ralph/lib/table/templatetags/table.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            if attr._meta.get_field(elem).choices:
+                return getattr(attr, 'get_{}_display'.format(elem))()
+            attr = getattr(attr, elem)
+        except AttributeError:
+            return None
+    return attr
+
+
+@register.inclusion_tag('table.html')
+def table(queryset, *fields):
```
Comment (mkurek) [inline src/ralph/lib/table/templatetags/table.py]: missing docstring

File: src/ralph/accounts/templates/ralphuser/user_additional_info.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{% extends BASE_TEMPLATE %}
+{% load i18n table %}
+{% block view_content %}
+<div class="large-12 columns">
+  <h1>{{ object.first_name }} {{ object.last_name }}</h1>
+  <br /> <br />
+  <h5>{% trans 'Assigned assets' %}:</h5>
+
+  {% table asset_list 'id' 'model__category' 'model__manufacturer' 'model__name' 'sn' 'barcode' 'remarks' 'status' %}
```
Comment (mkurek) [inline src/ralph/accounts/templates/ralphuser/user_additional_info.html]: imo it'd be better to define this list in python (somewhere in `RalphDetailView`). @ar4s what do you think?

Comment (ar4s) [inline src/ralph/accounts/templates/ralphuser/user_additional_info.html]: :+1:

File: src/ralph/lib/table/table.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            try:
+                name = getattr(self, field).title
+            except AttributeError:
+                name = get_field_by_relation_path(
+                    self.queryset.model, field
+                ).verbose_name
+            headers.append(name)
+        return headers
+
+    def get_value(self, item, field):
```
Comment (vi4m) [inline src/ralph/lib/table/table.py]: rename to `get_field_value` and think again about better docstring!

File: src/ralph/lib/table/table.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+class TableRender(object):
+
+    """
+    Generate HTML tables for the defined configuration.
+
+    Example:
+        >>> table = TableRender(queryset, ['id', 'name'])
+        >>> table.render()
+        [['id', 'name'], ['1', 'test']]
```
Comment (vi4m) [inline src/ralph/lib/table/table.py]: is this really rendered output? I would think it will generate html ?

File: src/ralph/lib/table/table.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            colspan = len(self.list_display)
+        for item in self.queryset.values(*list_display):
+            result.append([
+                {'value': self.get_field_value(item, field), 'html': ''}
+                for field in self.list_display
+            ])
+            if self.additional_row_method:
+                for method in self.additional_row_method:
+                    additional_data = [
+                        {'value': i, 'html': ' colspan={}'.format(colspan)}
```
Comment (ar4s) [inline src/ralph/lib/table/table.py]: try use `from django.forms.utils import flatatt`

File: src/ralph/lib/table/templates/table.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+      {% for header_name in headers %}
+        <th>{{ header_name|title }}</th>
+      {% endfor %}
+    </tr>
+  </thead>
+  <tbody>
+  {% for items in rows %}
+    <tr>
+      {% for item in items %}
+        <td{{ item.html }}>
```
Comment (mkurek) [inline src/ralph/lib/table/templates/table.html]: don't you think that `html_attributes` fit better here?

File: src/ralph/lib/table/table.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    Choices = None
+    use_choices = False
+
+from django.core.exceptions import FieldDoesNotExist
+from django.forms.utils import flatatt
+
+from ralph.admin.helpers import get_field_by_relation_path
+
+
+class TableRenderer(object):
```
Comment (mkurek) [inline src/ralph/lib/table/table.py]: It's not renderer anymore (it isn't rendering anything) - think about better name

File: src/ralph/lib/table/table.py [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+    """
+    Generate HTML tables for the defined configuration.
+
+    Example:
+        >>> table = TableRender(queryset, ['id', 'name'])
+        >>> table.render()
```
Comment (mkurek) [inline src/ralph/lib/table/table.py]: you mean `get_table_content`?

File: src/ralph/lib/table/table.py [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+        """
+        Initialize table class
+
+        :param queryset: Django queryset
+        :param list_display: Field list to display
+        :param additional_row_method: list of additional method for each row
+        """
+        self.queryset = queryset
+        self.list_display = list_display
+        self.additional_row_method = additional_row_method
```
Comment (mkurek) [inline src/ralph/lib/table/table.py]: what if `additional_row_method` is not passed (you are iterating over it in `get_table_content`)? Shouldn't there be `or []`?

**Comment by ar4s** [comment]
:ok:

**Comment by mkurek** [comment]
:ok:

**Pull request merged**